### PR TITLE
Add examples of linking and WASI

### PR DESCRIPTION
**Pull request by alexcrichton**
This commit adds two example programs, one for linking two modules
together and one for instantiating WASI. The linkage example
additionally uses WASI to get some meaningful output at this time.

cc #1272

### CODE REVIEW

File: examples/linking.c [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+    printf("> Error loading module!\n");
+    exit(1);
+  }
+  fclose(file);
+
+  // Parse the wat into the binary wasm format
+  wasm_byte_vec_t error;
+  if (wasmtime_wat2wasm(engine, &wat, bytes, &error) == 0) {
+    fprintf(stderr, "failed to parse wat %.*s\n", (int) error.size, error.data);
+    exit(0);
```
Comment (sunfishcode) [inline examples/linking.c]: This should exit with a non-zero status.

**Comment by alexcrichton** [comment]
Good idea! I've added all the examples as pages to the book, @sunfishcode want to take another look?

File: docs/embed-rust.md [hunk 2]
```diff
@@ -3,7 +3,7 @@
 This document shows an example of how to embed Wasmtime using the [Rust
 API][apidoc] to execute a simple wasm program. Be sure to also check out the
 [full API documentation][apidoc] for a full listing of what the [`wasmtime`
-crate][crate] has to offer.
+crate][wasmtime] has to offer.
```
Comment (sunfishcode) [inline docs/embed-rust.md]: Would it make sense to mention the examples here too, just to make sure people who skip ahead to the embedding chapter don't miss the examples that go along with it?

Comment (alexcrichton) [reply docs/embed-rust.md]: Sure!

**Pull request merged**